### ISSUE TITLE
String/Bytes utilities: contains, chomp, readline?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#18849]
While working with strings this weekend, I found myself wondering whether we could/should support the following capabilities:

* a `.contains()` query that would be much like `find()`, but would just return true or false, similar to `.contains()` on other collections.  This would support `if fullname.contains("Brad")` rather than `if fullname.find("Brad") != -1`
* a `.chomp()` method as in Python and Ruby that removes the newline from the end of a string
* an optional boolean argument on `readline()` that says whether to preserve the newline or drop it

Note that, for my sake, either of the second or third bullet would have been sufficient—I wouldn't necessarily need both.

**Comment by mppf** [comment]
Regarding chomp, we have this https://chapel-lang.org/docs/builtins/String.html#String.string.strip which is more general but might fit the bill. AFAIK Python itself does not have `chomp` but people use `rstrip` or `strip` instead.

> a .readline() call that doesn't require being called on a channel (i.e., uses stdin)

`stdin` is a channel, right? Are you just asking for a top-level function form (i.e. not a method) like `writeln` has ?

**Comment by bradcray** [comment]
> Regarding chomp, we have this https://chapel-lang.org/docs/builtins/String.html#String.string.strip which is more general but might fit the bill. AFAIK Python itself does not have chomp but people use rstrip or strip instead.

Yeah, this is what I've been using, but it feels heavyweight, and or like a big hammer, compared to `chomp()`.  The ability to have `readline()` drop the linefeed would probably be best of all, because then it would create the string I want from the start (vs. creating a string and then creating a second copy of it that's `size-1` big).  But `chomp()` still seems attractive from a simplicity / familiarity standpoint.  Similar to how I can use `.find()` instead of `.contains()` but `.contains()` feels simpler and reads more nicely for the common "does it contain?" case.

> stdin is a channel, right? Are you just asking for a top-level function form (i.e. not a method) like writeln has ?

Exactly, sorry to be unclear.  I've edited the OP to try and clarify.

**Comment by bradcray** [comment]
@mppf: You gave it a thumbs-up, so I assume don't object, but just to double-check:  Any reason you can think of that we only supported .readline() on a channel and not as a standalone procedure?  (or that we wouldn't want to do so).  I'd forgotten this, but it looks like this got added as part of the great QIO merge(?).

**Comment by mppf** [comment]
Right, a top-level `readline` just uses `stdin` would be fine with me. (I would not want it included by default though, because I would want it to throw, similar to the discussion in #19131). I also think it's OK to have people write `stdin.readline()` and I don't think that's an undue burden when working with `stdin`.

**Comment by mppf** [comment]
Yes, it would be reasonable to either add a standalone readline or make the other read functions available as e.g. `stdin.readln()`.
